### PR TITLE
[Non-Editable Folder Plugin] Created new non-editable folder plugin

### LINKED ISSUE #3612: [Folder Type] Need a non-editable folder type plugin
Currently all folders are creatable/editable, so any providers that want to use the folder type don't have much of a say in creatability/editability this has caused persistence issues: see #2634 

Need a non-editable folder plugin that will create a noneditable.folder type.
Need to also update list and folder views to handle this new folder type.

## Testing
- need to have a programmatically created noneditable.folder 
- test that you cannot create from the create menu
- test that you can not "Edit Properties"
- test that you can not edit the name in the Browse Bar

### PR DESCRIPTION
**Pull request by jvigliotta**
This will fix issues where folders from various sources which are not supposed to be editable will now NOT be editable. No 'edit properties' or name editing in browse bar will be available for folders that use this new type.

closes #3612 

### Author Checklist
Check | Passed
-- | --
Changes address original issue? | Y
Unit tests included and/or updated with changes? | Y
Command line build passes? | Y
Changes have been smoke-tested? | Y
Testing instructions included? | Y

### CODE REVIEW

File: src/plugins/folderView/FolderListView.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ define([
             name: 'List View',
             cssClass: 'icon-list-view',
             canView: function (domainObject) {
-                return domainObject.type === 'folder';
+                return domainObject.type === 'folder' || domainObject.type === 'noneditable.folder';
```
Comment (deeptailor) [inline src/plugins/folderView/FolderListView.js:38]: can we change this to be an array of types, declared at the top of the file? We may need to add new types that can use this view in the future

File: src/plugins/folderView/FolderGridView.js [hunk 2]
```diff
@@ -33,7 +33,7 @@ define([
             name: 'Grid View',
             cssClass: 'icon-thumbs-strip',
             canView: function (domainObject) {
-                return domainObject.type === 'folder';
+                return domainObject.type === 'folder' || domainObject.type === 'noneditable.folder';
```
Comment (deeptailor) [inline src/plugins/folderView/FolderGridView.js:36]: Same comment as below about having an array of types

**Review by deeptailor — CHANGES_REQUESTED** [review]
Just one comment, otherwise looks good

File: src/plugins/folderView/FolderGridView.js (hunk 2 above)
Comment (jvigliotta) [reply src/plugins/folderView/FolderGridView.js]: Added a constants file to use in both.

**Review by deeptailor — APPROVED** [review]
LGTM

**Comment by jvigliotta** [comment]
@deeptailor can we merge this one?

**Pull request merged**